File: app/src/main/java/io/github/joseluiscd/seagull/adapters/TrackViewHolder.java
package io.github.joseluiscd.seagull.adapters;

import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import io.github.joseluiscd.seagull.R;
import io.github.joseluiscd.seagull.model.Track;

/**
 * Created by joseluis on 11/01/18.
 */

public class TrackViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener, View.OnLongClickListener {
    private final View mView;
    TextView title;
    TextView artist;
    TextView album;
    ImageView albumArt;

    private ClickListener clickListener;

    public TrackViewHolder(View view, ClickListener clickListener) {
        super(view);
        view.setOnClickListener(this);
        view.setOnLongClickListener(this);

        mView = view;
        title = view.findViewById(R.id.track_title);
        artist = view.findViewById(R.id.track_artist);
        album = view.findViewById(R.id.track_album);
        albumArt = view.findViewById(R.id.track_album_image);

        this.clickListener = clickListener;
    }

    @Override
    public void onClick(View v) {
        if(clickListener != null) clickListener.onClick(mView, getAdapterPosition());
    }

    @Override
    public boolean onLongClick(View v) {
        if(clickListener != null) clickListener.onLongClick(mView, getAdapterPosition());
        return true;
    }
}

File: app/src/main/java/io/github/joseluiscd/seagull/adapters/TrackViewHolder.kt
package io.github.joseluiscd.seagull.adapters

import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.ContextMenu
import android.view.Menu
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import com.squareup.picasso.Picasso

import io.github.joseluiscd.seagull.R
import io.github.joseluiscd.seagull.model.Track

/**
 * Created by joseluis on 11/01/18.
 */

class TrackViewHolder(
        private val mView: View,
        private val clickListener: ClickListener? = null,
        private val menuListener: MenuListener? = null
)
    : AbstractViewHolder<Track>(mView, clickListener, menuListener) {

    internal var title: TextView
    internal var artist: TextView
    internal var album: TextView
    internal var albumArt: ImageView

    init {
        title = mView.findViewById(R.id.track_title)
        artist = mView.findViewById(R.id.track_artist)
        album = mView.findViewById(R.id.track_album)
        albumArt = mView.findViewById(R.id.track_album_image)
    }

    override fun fillItem(t: Track) {
        title.text = t.title
        artist.text = t.artist
        album.text = t.album

        Picasso.with(albumArt.context.applicationContext)
                .load(t.albumArtURL)
                .placeholder(R.drawable.lp)
                .into(this.albumArt)
    }
}
